lib/src/signed_video_h26x_auth.c

```diff
@@ -522,7 +688,7 @@ validate_authenticity(signed_video_t *self)
     verify_success = verify_hashes_without_sei(self);
   } else {
     if (self->gop_info->signature_hash_type == DOCUMENT_HASH) {
-      verify_success = verify_hashes_with_hash_list(self, &num_expected_nalus, &num_received_nalus);
+      verify_success = verify_document_gop_hash(self, &num_expected_nalus, &num_received_nalus);
     } else {
       verify_success = verify_hashes_with_gop_hash(self, &num_expected_nalus, &num_received_nalus);
```

Comment:
Can this happen at all now? I think it is good to keep it until we remove the old replaced code, but if it cannot happen you can add an `assert(false);` to catch an unexpected event.